RN0.58 Refactor NativeModuleRegistryBuilder remove addNativeModule

[NativeModuleRegistryBuilder](https://github.com/facebook/react-native/commit/875f70b8b691cd1fb3e802594bbd4461b9959751#diff-9dc4073d6038a2b98f9f36a9233be809l)

NativeModuleRegistryBuilder remove addNativeModule method 

[ReactContextBuilder line 137 error](https://github.com/joltup/react-native-threads/blob/master/android/src/main/java/com/reactlibrary/ReactContextBuilder.java)

Is there any progress on this? 
Compiled with `"react-native": "^0.58.4"` and `"react-native-threads": "0.0.15"`.
Note that I also tried to downgrade react-native to `0.57.8`, but then threads did not compilel due to constructor parameters mismatch.

```
> Task :react-native-threads:compileDebugJavaWithJavac FAILED
/...project.../node_modules/react-native-threads/android/src/main/java/com/reactlibrary/ReactContextBuilder.java:137: error: cannot find symbol
                nativeRegistryBuilder.addNativeModule(nativeModule);
```

--- a/android/src/main/java/com/reactlibrary/ReactContextBuilder.java
+++ b/android/src/main/java/com/reactlibrary/ReactContextBuilder.java
@@ -8,7 +8,7 @@ import com.facebook.react.ReactPackage;
 import com.facebook.react.bridge.CatalystInstance;
 import com.facebook.react.bridge.CatalystInstanceImpl;
 import com.facebook.react.bridge.JSBundleLoader;
-import com.facebook.react.bridge.JSCJavaScriptExecutorFactory;
+import com.facebook.react.jscexecutor.JSCExecutorFactory;
 import com.facebook.react.bridge.JavaScriptExecutor;
 import com.facebook.react.bridge.NativeModuleCallExceptionHandler;
 import com.facebook.react.bridge.ReactApplicationContext;
@@ -58,7 +58,7 @@ public class ReactContextBuilder {
     public ReactApplicationContext build() throws Exception {
         String appName = Uri.encode(parentContext.getPackageName());
         String deviceName = Uri.encode(getFriendlyDeviceName());
-        JavaScriptExecutor jsExecutor = new JSCJavaScriptExecutorFactory(appName, deviceName)
+        JavaScriptExecutor jsExecutor = new JSCExecutorFactory(appName, deviceName)
                 .create();
 
         // fresh new react context

--- a/android/src/main/java/com/reactlibrary/ThreadSelfModule.java
+++ b/android/src/main/java/com/reactlibrary/ThreadSelfModule.java
@@ -3,9 +3,12 @@ package com.reactlibrary;
 import com.facebook.react.bridge.ReactApplicationContext;
 import com.facebook.react.bridge.ReactContextBaseJavaModule;
 import com.facebook.react.bridge.ReactMethod;
+import com.facebook.react.module.annotations.ReactModule;
 import com.facebook.react.modules.core.DeviceEventManagerModule;
 
+@ReactModule(name = ThreadSelfModule.REACT_MODULE_NAME)
 public class ThreadSelfModule extends ReactContextBaseJavaModule {
+    public static final String REACT_MODULE_NAME = "ThreadSelfManager";
 
     private int threadId;
     private ReactApplicationContext parentContext;
@@ -21,7 +24,7 @@ public class ThreadSelfModule extends ReactContextBaseJavaModule {
 
     @Override
     public String getName() {
-        return "ThreadSelfManager";
+        return REACT_MODULE_NAME;
     }
 
     @ReactMethod

